tests/cancun/eip1153_tstore/test_tstore_reentrancy.py

```diff
@@ -0,0 +1,212 @@
+            code=Switch(
+                cases=[
+                    Case(
+                        condition=Op.EQ(Op.CALLDATALOAD(0), do_tstore),
+                        action=subcall_code,
+                    ),
+                    Case(
+                        condition=Op.EQ(Op.CALLDATALOAD(0), do_reenter),
+                        action=Op.TSTORE(0, tload_value_set_before_call)
+                        + Op.SSTORE(slot_tload_before_call, Op.TLOAD(0))
+                        + Op.MSTORE(0, do_tstore)
+                        + Op.MSTORE(32, 0xFF)
+                        + Op.SSTORE(slot_subcall_worked, make_call(call_type))
+                        + Op.SSTORE(slot_tload_in_subcall_result, Op.MLOAD(32))
+                        + Op.SSTORE(slot_tload_after_call, Op.TLOAD(0))
+                        + Op.SSTORE(slot_tload_1_after_call, Op.TLOAD(1))
+                        + Op.TSTORE(0, 50)
+                        + Op.SSTORE(slot_tstore_overwrite, Op.TLOAD(0))
+                        + Op.SSTORE(slot_code_worked, 1),
+                    ),
+                ],
+                default_action=None,
+            ),
+            storage={
+                slot_tload_before_call: 0xFF,
+                slot_tload_in_subcall_result: 0xFF,
+                slot_tload_after_call: 0xFF,
+                slot_subcall_worked: 0xFF,
+                slot_tload_1_after_call: 0xFF,
+                slot_tstore_overwrite: 0xFF,
+                slot_code_worked: 0xFF,
+            },
+        ),
+        address_code: Account(
+            balance=0,
+            nonce=0,
+            code=subcall_code,
+            storage={},
+        ),
+        TestAddress: Account(
+            balance=7000000000000000000,
+            nonce=0,
+            code="0x",
+            storage={},
+        ),
+    }
+
+    post: Dict[Address, Union[Account, object]] = {}
+
+    def failing_calls() -> bool:
+        return call_type == Op.STATICCALL or call_return == Op.REVERT or call_return == Om.OOG
+
+    def successful_delegate_or_callcode() -> bool:
+        return (
+            (call_type == Op.DELEGATECALL or call_type == Op.CALLCODE)
+            and call_return != Op.REVERT
+            and call_return != Om.OOG
+        )
+
+    if call_dest == Op.ADDRESS():
+        # if reentrancy
+        post[address_to] = Account(
+            storage={
+                slot_code_worked: 1,
+                slot_tload_before_call: tload_value_set_before_call,
+                slot_tload_in_subcall_result: (
+                    # we fail to obtain in call result if it fails
+                    0xFF
+                    if call_type == Op.STATICCALL or call_return == Om.OOG
+                    else tload_value_set_in_call
+                ),
+                # reentrant tstore overrides value in upper level
+                slot_tload_after_call: (
+                    tload_value_set_before_call if failing_calls() else tload_value_set_in_call
```

Comment:
No need to call if converted to variable.
```suggestion
                    tload_value_set_before_call if failing_calls else tload_value_set_in_call
```